nexteuropa_dashboard_agent/nexteuropa_dashboard_agent.module

```diff
@@ -79,26 +79,32 @@ function nexteuropa_dashboard_agent_access_callback() {
       }
       else {
         $allowed_ips = preg_replace('/\s+/', '', $allowed_ips);
-        $allowed_ip_array = explode('-', $allowed_ips);
-        list($lower, $upper) = $allowed_ip_array;
-        $lower_dec = (float) sprintf("%u", ip2long($lower));
-        $upper_dec = (float) sprintf("%u", ip2long($upper));
-        $ip_dec = (float) sprintf("%u", ip2long(ip_address()));
-        if (($ip_dec >= $lower_dec) && ($ip_dec <= $upper_dec)) {
-          return TRUE;
-        }
-        else {
-          watchdog(
-            'nexteuropa_dashboard_agent',
-            "Requester IP *** %requester_ip *** is not allowed. Should be in the range *** %allowed_ips ***.",
-            array(
-              '%requester_ip' => ip_address(),
-              '%allowed_ips' => $allowed_ips,
-            ),
-            WATCHDOG_WARNING
-          );
-          return FALSE;
+        $allowed_ip_list = explode(',', $allowed_ips);
+        // Compare current IP with the list of allowed IPs.
+        foreach ($allowed_ip_list as $allowed_ip) {
+          $allowed_ip_array = explode('-', $allowed_ip);
+          list($lower, $upper) = $allowed_ip_array;
```

Comment:
If the configuration provide a single ip, your $upper_dec ip will be empty, causing the condition of line 90 to fail.

Reply:
Good remark @jonhy81 
This is now solved.